stubs/pyxdg/xdg/Menu.pyi

```diff
@@ -0,0 +1,168 @@
+    Filename: str
+    Prefix: str
+    DesktopFileID: str
+    def setAttributes(self, filename: str, dir: str = "", prefix: str = "") -> None: ...
+    def updateAttributes(self) -> None: ...
+    def __cmp__(self, other: MenuEntry) -> int: ...
+    def __lt__(self, other: MenuEntry) -> bool: ...
+    def __eq__(self, other: object) -> bool: ...
+
+class Separator:
+    Parent: Menu
+    Show: bool
+    def __init__(self, parent: Menu) -> None: ...
+
+class Header:
+    Name: str
+    GenericName: str
+    Comment: str
+    def __init__(self, name: str, generic_name: str, comment: str) -> None: ...
+
+TYPE_DIR: Literal[0] = 0
+TYPE_FILE: Literal[1] = 1
+
+class XMLMenuBuilder:
+    debug: bool
+    def __init__(self, debug: bool = False) -> None: ...
+    cache: MenuEntryCache
+    def parse(self, filename: str | None = None) -> Menu: ...
+    def parse_menu(self, node: xml.dom.Node, filename: str): ...
+    def parse_node(self, node: xml.dom.Node, filename: str, parent: Menu | None = None) -> None: ...
+    def parse_layout(self, node: xml.dom.Node): ...
+    def parse_move(self, node: xml.dom.Node): ...
+    def parse_rule(self, node: xml.dom.Node): ...
+    def parse_bool_op(self, node: xml.dom.Node, operator: ast.And | ast.Or): ...
+    def parse_rule_node(self, node: xml.dom.Node): ...
+    def parse_app_dir(self, value: str, filename: str, parent: str) -> None: ...
+    def parse_default_app_dir(self, filename: str, parent: str) -> None: ...
+    def parse_directory_dir(self, value: str, filename: str, parent: str) -> None: ...
+    def parse_default_directory_dir(self, filename: str, parent: str) -> None: ...
+    def parse_merge_file(self, value: str, child: Menu | MenuEntry, filename: str, parent: str) -> None: ...
+    def parse_merge_dir(self, value: str, child: Menu | MenuEntry, filename: str, parent: str) -> None: ...
+    def parse_default_merge_dirs(self, child, filename: str, parent: str) -> None: ...
+    def merge_file(self, filename: str, child: Unused, parent: Menu) -> None: ...
+    def parse_legacy_dir(self, dir_: str, prefix: str, filename: str, parent: str) -> None: ...
+    def merge_legacy_dir(self, dir_: str, prefix: str, filename: str, parent: str): ...
+    def parse_kde_legacy_dirs(self, filename: str, parent: str) -> None: ...
+    def post_parse(self, menu: Menu) -> None: ...
+    def generate_not_only_allocated(self, menu: Menu) -> None: ...
+    def generate_only_allocated(self, menu: Menu) -> None: ...
+    def handle_moves(self, menu: Menu) -> None: ...
+
+class MenuEntryCache:
+    cacheEntries: dict[str, list[MenuEntry]]
+    cache: dict[str, list[MenuEntry]]
+    def __init__(self) -> None: ...
+    def add_menu_entries(self, dirs: Iterable[str], prefix: str = "", legacy: bool = False) -> None: ...
+    def get_menu_entries(self, dirs: Collection[str], legacy: bool = True) -> list[MenuEntry]: ...
+
+def parse(filename: AnyStr | None = None, debug: bool = False) -> XMLMenuBuilder: ...
```

Comment:
`AnyStr` looks incorrect here -- it's only used once in the signature, so it's not a generic function. That means using a `TypeVar` has little meaning here; you should either use `str`, `bytes` or `str | bytes`